client/asset/dcr/rpcwallet.go

```diff
@@ -642,9 +723,13 @@ func (w *rpcWallet) PeerCount(ctx context.Context) (uint32, error) {
 
 // AddressPrivKey fetches the privkey for the specified address.
 // Part of the Wallet interface.
-func (w *rpcWallet) AddressPrivKey(ctx context.Context, address stdaddr.Address) (*dcrutil.WIF, error) {
-	wif, err := w.client().DumpPrivKey(ctx, address)
-	return wif, translateRPCCancelErr(err)
+func (w *rpcWallet) AddressPrivKey(ctx context.Context, address stdaddr.Address) (*secp256k1.PrivateKey, error) {
+	wif, err := w.rpcClient.DumpPrivKey(ctx, address)
+	if err != nil {
+		return nil, translateRPCCancelErr(err)
+	}
+	priv := secp256k1.PrivKeyFromBytes(wif.PrivKey())
```

Comment:
I know this was like this before, but it might be prudent to do the parsing manually so we can validate the private key bytes.

```go
var priv PrivateKey
if overflow := priv.Key.SetByteSlice(privKeyBytes); overflow || priv.Key.IsZero() {
	return nil, errors.New("invalid private key")
}
```

as described in the docs here https://pkg.go.dev/github.com/decred/dcrd/dcrec/secp256k1/v4#PrivKeyFromBytes